### ISSUE TITLE
mat2surf

### ISSUE THREAD

**Issue by HappyKerry** [firekind/mat-to-surf#1]
Haveyou fix the issue?I want to use mat2surf before nvinfer inference transform,and replace the in_surf,so we can add custom pre process like warppersfective,is it the right way?

**Comment by firekind** [comment]
Nope. Couldn't figure out why the issue was occurring. Everything works fine on dGPU, the problem still occurs on jetson. 
If you are using deepstream 6+, you can try using the nvdspreprocess plugin (although I have not used it).

Another approach could be to use unified memory type of NvBufSurface, map it to opencv mat (using the code in dsexample), and carefully apply opencv functions. I say carefully since depending on what you are doing, opencv might create a new cv::Mat internally and give that to you, instead of using the cv::Mat that points to the unified memory (this mostly happens when the operation you are performing changes the dimensions of the image). This approach does work on dGPU, have not checked for memory leak on jetson though.
